bgpd/bgp_vty.c

```diff
@@ -8981,6 +8981,53 @@ ALIAS (af_rd_vpn_export,
        "Between current address-family and vpn\n"
        "For routes leaked from current address-family to vpn\n")
 
+DEFPY(af_label_vpn_export_allocation_mode,
+      af_label_vpn_export_allocation_mode_cmd,
+      "[no$no] label vpn export allocation-mode <per-vrf$label_per_vrf|per-nexthop$label_per_nh>",
+      NO_STR
+      "label value for VRF\n"
+      "Between current address-family and vpn\n"
+      "For routes leaked from current address-family to vpn\n"
+      "Label allocation mode\n"
+      "Allocate one label for all BGP updates of the VRF\n"
+      "Allocate a label per connected next-hop in the VRF\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	afi_t afi;
+
+	afi = vpn_policy_getafi(vty, bgp, false);
+
+	/* no change */
+	if (((!no && label_per_nh) || (no && label_per_vrf)) &&
+	    CHECK_FLAG(bgp->vpn_policy[afi].flags,
+		       BGP_VPN_POLICY_TOVPN_LABEL_PER_NEXTHOP))
+		return CMD_SUCCESS;
+	if (((!no && label_per_vrf) || (no && label_per_nh)) &&
+	    !CHECK_FLAG(bgp->vpn_policy[afi].flags,
+			BGP_VPN_POLICY_TOVPN_LABEL_PER_NEXTHOP))
+		return CMD_SUCCESS;
+
+	/*
+	 * pre-change: un-export vpn routes (vpn->vrf routes unaffected)
+	 */
+	vpn_leak_prechange(BGP_VPN_POLICY_DIR_TOVPN, afi, bgp_get_default(),
+			   bgp);
+
+	if ((!no && label_per_nh) || (no && label_per_vrf))
+		SET_FLAG(bgp->vpn_policy[afi].flags,
+			 BGP_VPN_POLICY_TOVPN_LABEL_PER_NEXTHOP);
+	if ((!no && label_per_vrf) || (no && label_per_nh))
+		UNSET_FLAG(bgp->vpn_policy[afi].flags,
```

Comment:
idem

Reply:
not sure it is possible to do that.
each configuration change performs a refresh of the BGP exportation before and after config

Reply:
OK